src/classify/classify.ts

```diff
@@ -1,9 +1,77 @@
-import { ensureLengthAtLeast, loadHTML, parseText } from "../utils";
-import { CatalogEntryType, TypedCatalogEntry } from "./types";
+import assert from "assert";
+import { College } from "../urls";
+import {
+  ensureAtLeastLength,
+  loadHTML,
+  majorNameToFileName,
+  parseText,
+} from "../utils";
+import { CatalogEntryType, FilterError, type TypedCatalogEntry } from "./types";
+import { mkdir, writeFile } from "fs/promises";
+import { join } from "path";
 
-export const addTypeToUrl = async (url: URL): Promise<TypedCatalogEntry> => {
-  const type = getUrlType(await loadHTML(url.href));
-  return { url, type };
+export const classify = async (
+  url: URL,
+  filterTypes: CatalogEntryType[],
+): Promise<TypedCatalogEntry> => {
+  const html = await loadHTML(url.href);
+
+  const { degreeType, yearVersion, college, majorName } = getMetadata(
+    html,
+    url,
+  );
+
+  if (!filterTypes.includes(degreeType)) {
+    throw new FilterError(degreeType, filterTypes);
+  }
+
+  const savePath = join(
+    "degrees",
+    degreeType,
+    yearVersion.toString(),
+    college,
+    majorNameToFileName(majorName),
+  );
+  await mkdir(savePath, { recursive: true });
+  await writeFile(`${savePath}/html.html`, html.html());
+
+  return { url, degreeType, yearVersion, college, majorName, savePath, html };
+};
+
+const getCollegeFromURL = (url: URL): College => {
+  const isArchived = url.href.includes("archive");
+  const college = url.toString().split("/")[isArchived ? 6 : 4] as College;
```

Comment:
that makes sense, but this is just a temporary solution for now since I want to get the college straight from the `getMajorLinks` step instead of here (we build the URL from that step so we have the metadata already)